cipd_packages/device_doctor/lib/src/android_device.dart

```diff
@@ -453,14 +453,20 @@ class AndroidDevice implements Device {
     processManager ??= LocalProcessManager();
     final int timeoutSecs = 60;
     print('Device recovery: deleting package caches...');
-    await eval('adb', <String>['wait-for-device'], canFail: false, processManager: processManager)
+    await eval('adb',
+            <String>['wait-for-device', 'shell', 'while [[ -z \$(getprop sys.boot_completed) ]]; do sleep 1; done;'],
+            canFail: false, processManager: processManager)
         .timeout(Duration(seconds: timeoutSecs));
     await deletePackageCache();
```

Comment:
Are these calls restarting the device?

Reply:
Aside from the reboot, it should be. I think the issue is the adb server and whats calling this script vs other things on the bot.

Reply:
The delete calls don't restart the device. I guess the other question in regards to those is how does it wait for the package to be deleted and how long does it take to delete a package?

Reply:
This is hapepning before the package delete call itself

Reply:
Further up in the call stack it attempts to list devices and if it cannot it kills the adb server for some reason.
https://github.com/flutter/cocoon/blob/main/cipd_packages/device_doctor/lib/src/android_device.dart#L58